aml_module/invoker.py

```diff
@@ -36,67 +34,92 @@ def is_timestamp_ascending(timestamps):
     return 0
 
 
-def invoke(input_path, detect_mode, timestamp_column, value_column, batch_size, threshold, sensitivity,
-            appendMode, compute_stats_in_visualization, output_path):
-    data_frame_directory = load_data_frame_from_directory(input_path)
+def read_as_dataframe(input_path: str):
+    if os.path.isfile(input_path):
+        if input_path.endswith(".csv"):
+            return pd.read_csv(input_path)
+        elif input_path.endswith(".parquet"):
+            return pd.read_parquet(input_path)
+    else:
+        dir_path = pathlib.Path(input_path)
+
+        csv_files = list(dir_path.glob("**/*.csv"))
+        if csv_files:
+            df_from_csv_files = (pd.read_csv(f) for f in csv_files)
+            return pd.concat(df_from_csv_files, ignore_index=True)
+
+        parquet_files = list(dir_path.glob("**/*.parquet"))
+        if parquet_files:
+            df_from_parquet_files = (pd.read_parquet(f) for f in parquet_files)
+            return pd.concat(df_from_parquet_files, ignore_index=True)
```

Comment:
why do we need these complicated data processing steps?

Reply:
And also, more support to the case that multiple files in a folder can be processed. These goals are raised by @weedqian, maybe he can explain more about why we choose such input types.

Reply:
As there are scenarios with this kind of data source, just keep these codes.